pyron-plugin-impl/src/test/resources/plugins/transformer/rules.json

```diff
@@ -249,6 +249,27 @@
             }
           ]
         },
+        {
+          "method": "GET",
+          "pathPattern": "/dyn-header-can-use-pattern-parameters-with-defined-size",
+          "requestPlugins": [
+            {
+              "name": "transform-request",
+              "conf": {
+                "body": {},
+                "headers": {
+                  "set": {
+                    "X-Account": {
+                      "path": "$body.scp",
+                      "pattern": "account-{countryCode_0_3}{a_4}-{b_4}-{c_4}-{d_4}",
```

Comment:
Please add docs in `https://github.com/cloudentity/pyron/blob/dev/docs/plugins/transform-request.md`. Can we replace `_` with `:`? It seems safer keyword, because underscore are often used in param names, e.g. param_name. It might be a surprise to the user. 

Pyron should fail to start if the config for `X-Account` is invalid, would it? The validation logic of transform-request plugin config is put in circe decoders (see https://github.com/cloudentity/pyron/blob/dev/pyron-plugin-impl/src/main/scala/com/cloudentity/pyron/plugin/impl/transformer/domain.scala and https://github.com/cloudentity/pyron/blob/dev/pyron-plugin/src/main/scala/com/cloudentity/pyron/plugin/util/value/domain.scala)

Reply:
Before this change underscore was 'effectively' not allowed, because it mapped directly to capture-group name, which does not accept '_' it. I agree people like to use _ in param names, so I took the chance while dropping dependency on named capture groups to allow it. We now refer to all capture groups by index, names are resolved to index in PreparedRewrite setup. These ranges come handy if matching chunks are not separated by '/' to avoid eager capture of everything. I didn't want to extend param names with some special chars valid only for special scenarios, but accepting ':' for defining ranges seems ok. I'll look into it.